inspirehep/modules/orcid/utils.py

```diff
@@ -54,3 +60,378 @@ def get_authors_credentials(author):
+                record['external-ids']['external-id'][0]['external-id-value'])
+        if record['external-ids']['external-id'][0]['external-id-type'] == 'arxiv':
+            orcid_records_arxivs.append(
+                record['external-ids']['external-id'][0]['external-id-value'])
+
+    return orcid_records_dois, orcid_records_arxivs
+
+
+def extract_author_profiles_from_hep_records(records):
+    possible_profiles = set()
+    for record in records:
+        try:
+            for author in record['inner_hits']['authors']['hits']['hits']:
+                possible_profiles.add(author['_source']['record']['$ref'])
+        except KeyError:
+            continue
+    return possible_profiles
+
+
+def solve_references_of_authors(refs):
+    possible_profiles = []
+    current_app.logger.info(refs)
+    for ref in refs:
+        author_profile = replace_refs({'$ref': ref}, source='es')
+        try:
+            # If the profile has an orcid it's already claimed by somebody.
+            for identifier in author_profile['ids']:
+                if 'orcid' in identifier:
+                    continue
+            possible_profiles.append(ref)
+        except KeyError:
+            continue
+    return possible_profiles
+
+
+def pick_records_from_author_profiles(profiles):
+    number_of_suggested_records = len(profiles)
+    number_of_records_per_profile = 1
+    picked_records = {}
+    picked_recids = []
+    if len(profiles) is 2:
+        number_of_suggested_records = 4
+        number_of_records_per_profile = 2
+    if len(profiles) > 10:
+        number_of_suggested_records = 10
+    for i, profile in enumerate(profiles):
+        picked_records[profile] = []
+        if i is number_of_suggested_records:
+            break
+        records = query_elasticsearch_for_literature_records_using_a_profile_reference(
+            profile)
+        for j, record in enumerate(records):
+            if j is number_of_records_per_profile:
+                break
+            if record['_source']['control_number'] not in picked_recids:
+                record['_source']['_profile'] = profile
+                picked_records[profile].append(record)
+                picked_recids.append(record['_source']['control_number'])
+    if len(picked_recids) is 1:
+        assign_profile(profile)
+    return picked_records
+
+
+def find_profiles(possible_profiles):
+    if len(possible_profiles) is 1:
+        profile = possible_profiles.pop()
+        assign_profile(profile)
+        return [profile]
+    if len(possible_profiles) is 0:
+        return None
+    chosen_records = pick_records_from_author_profiles(possible_profiles)
+    return chosen_records
+
+
+def assign_profile(profile, orcid=None):
+    if not orcid:
+        orcid = session['users_orcid']
+    author = query_for_author_profiles_using_ref(profile)[0]['_source']
+    author['ids'].append({
+        "type": u"ORCID",
+        "value": orcid
+    })
+    current_search_client.index(
```

Comment:
Yep, the change should totally go to the DB (and as a consequence it will be reflected into ES).
@jacquerie has just amended the schema WRT IDs.
https://github.com/inspirehep/inspire-next/blob/master/inspirehep/modules/records/jsonschemas/records/authors.json#L98